lib/ansible/modules/cloud/azure/azure_rm_cosmosdbaccount.py

```diff
@@ -0,0 +1,545 @@
+    from azure.mgmt.cosmosdb import CosmosDB
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMCosmosDBAccount(AzureRMModuleBase):
+    """Configuration class for an Azure RM Database Account resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str'
+            ),
+            kind=dict(
+                type='str',
+                choices=['global_document_db',
+                         'mongo_db',
+                         'parse']
+            ),
+            consistency_policy=dict(
+                type='dict',
+                options=dict(
+                    default_consistency_level=dict(
+                        type='str',
+                        choices=['eventual',
+                                 'session',
+                                 'bounded_staleness',
+                                 'strong',
+                                 'consistent_prefix']
+                    ),
+                    max_staleness_prefix=dict(
+                        type='int'
+                    ),
+                    max_interval_in_seconds=dict(
+                        type='int'
+                    )
+                )
+            ),
+            geo_rep_locations=dict(
+                type='list',
+                options=dict(
+                    name=dict(
+                        type='str',
+                        required=True
+                    ),
+                    failover_priority=dict(
+                        type='int',
+                        required=True
+                    )
+                )
+            ),
+            database_account_offer_type=dict(
+                type='str'
+            ),
+            ip_range_filter=dict(
+                type='str'
+            ),
+            is_virtual_network_filter_enabled=dict(
+                type='bool'
+            ),
+            enable_automatic_failover=dict(
+                type='bool'
+            ),
+            capabilities=dict(
+                type='list',
+                options=dict(
+                    name=dict(
+                        type='str',
+                        required=True
+                    )
+                )
+            ),
+            virtual_network_rules=dict(
+                type='list',
+                options=dict(
+                    id=dict(
+                        type='str',
+                        required=True
+                    )
+                )
+            ),
+            enable_multiple_write_locations=dict(
+                type='str'
```

Comment:
@zikalino thanks for submitting this module, I've been testing it and this should also be of type `bool` otherwise the changed check failed when comparing values

Reply:
@marc-sensenich thanks for testing it! I plan to make some updates and I plan to get this PR merged next week